wagtail/wagtailadmin/edit_handlers.py

```diff
@@ -586,15 +631,24 @@ def render_js_init(self):
         }))
 
 
-def InlinePanel(base_model, relation_name, panels=None, label='', help_text=''):
-    rel = getattr(base_model, relation_name).related
-    return type(str('_InlinePanel'), (BaseInlinePanel,), {
-        'relation_name': relation_name,
-        'related': rel,
-        'panels': panels,
-        'heading': label,
-        'help_text': help_text,  # TODO: can we pick this out of the foreign key definition as an alternative? (with a bit of help from the inlineformset object, as we do for label/heading)
-    })
+class InlinePanel(object):
+    def __init__(self, base_model, relation_name, panels=None, label='', help_text=''):
```

Comment:
Yep, deprecating the base_model parameter is the plan. I was thinking of just doing a simple check on len(args) - if there are two args, it's old style, if there's only one, it's new style.